libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -165,13 +163,9 @@ private synchronized static String generateUniqueId(String name, int length) {
         } else {
             String uniqueId;
             try {
-
-                // Uses SecureRandom to generate an AES-256 key.
-                final SecureRandom secureRandom = SecureRandom.getInstance(SHA1PRNG);
-
-                // SecureRandom does not require seeding. It's automatically seeded from system entropy.
+                // Create the key generator with its recommended secure random number generator provider algorithm.
                 final KeyGenerator keyGenerator = KeyGenerator.getInstance(AES);
-                keyGenerator.init(length, secureRandom);
+                keyGenerator.init(length);
```

Comment:
For `KeyGenerator`, the documentation in the description plus the Javadoc agreed it was recommended to let it have control over creation of the random number generator, its algorithm and seeding.